### PR TITLE
Support for flattened per-reporter staleness and its filters

### PR DESCRIPTION
**Pull request by kruai**
This follows up on #151. It adds `per_reporter_staleness` and `per_reporter_staleness_flat` to the schema, and filters on `per_reporter_staleness_flat` specifically. The intent is to get the regular object data in the response, while filtering on the flattened version like this:
```
filter: {
    per_reporter_staleness: {
        reporter: "yupana"
        stale_timestamp: {gte: "2020-02-10T08:07:03.354307Z"}
        last_check_in: {gte: "2020-02-09T08:07:03.354307Z"}
        check_in_succeeded: {is: true}
    }
}
```

### CODE REVIEW

**Review by thearifismail — APPROVED** [review]
LGTM

File: src/resolvers/hosts/hosts.integration.ts [hunk 1]
```diff
@@ -1105,6 +1131,56 @@ describe('hosts query', function () {
                 expect(data).toMatchSnapshot();
             });
         });
+
+        describe('per_reporter_staleness', function () {
+            test('reporter', async () => {
+                const { data } = await runQuery(PRS_QUERY,
+                    { filter: { per_reporter_staleness: { reporter: { eq: 'yupana' }}}});
+                data.hosts.data.should.have.length(2);
+                data.hosts.data[0].id.should.equal('6e7b6317-0a2d-4552-a2f2-b7da0aece49d');
```
Comment (fijshion) [inline src/resolvers/hosts/hosts.integration.ts:1140]: Only the system with id `f5ac67e1-ad65-4b62-bc27-845cc6d4bcee` has reporter=yupana. That's why this test is failing.

Comment (kruai) [reply src/resolvers/hosts/hosts.integration.ts:1140]: Oops - I didn't realize that my recent fix broke something else. I'll get that fixed ASAP

File: src/resolvers/hosts/hosts.integration.ts [hunk 2]
```diff
@@ -1105,6 +1131,56 @@ describe('hosts query', function () {
...
+                data.hosts.data[1].id.should.equal('f5ac67e1-ad65-4b62-bc27-845cc6d4bcee');
+            });
+
+            test('reporter and check in succeeded', async () => {
+                const { data } = await runQuery(PRS_QUERY,
+                    { filter: { per_reporter_staleness: { reporter: {eq: 'yupana'}, check_in_succeeded: {is: true}}}});
+                data.hosts.data.should.have.length(1);
+                data.hosts.data[0].id.should.equal('6e7b6317-0a2d-4552-a2f2-b7da0aece49d');
```
[the remark this answers is not in the document]
Comment (kruai) [reply src/resolvers/hosts/hosts.integration.ts:1148]: `check_in_succeeded=true`, but yeah that fixed it.

**Comment by kruai** [comment]
@fijshion no rush, but please take another look when you get a chance. I've only made a small update to `hosts.json` since your last review

**Review by fijshion — APPROVED** [review]

**Pull request merged**